build.gradle

```diff
@@ -334,17 +336,20 @@ dependencies {
     annotationProcessor "org.springframework.boot:spring-boot-configuration-processor"
 
     constraints {
-        implementation('org.bouncycastle:bcpg-jdk15on:1.68') {
-            because '1.66 has security vulnerabilities'
+        implementation('org.bouncycastle:bcpg-jdk18on:1.77') {
+            because 'jdk15on:1.70 has security vulnerabilities'
         }
-        implementation('org.bouncycastle:bcpkix-jdk15on:1.68') {
-            because '1.66 has security vulnerabilities'
+        implementation('org.bouncycastle:bcpkix-jdk18on:1.77') {
+            because 'jdk15on:1.70 has security vulnerabilities'
         }
-        implementation('org.bouncycastle:bcprov-ext-jdk15on:1.68') {
-            because '1.66 has security vulnerabilities'
+        implementation('org.bouncycastle:bcprov-ext-jdk18on:1.77') {
+            because 'jdk15on:1.70 has security vulnerabilities'
         }
-        implementation('org.bouncycastle:bcprov-jdk15on:1.68') {
-            because '1.66 has security vulnerabilities'
+        implementation('org.bouncycastle:bcprov-jdk18on:1.77') {
+            because 'jdk15on:1.70 has security vulnerabilities'
+        }
+        implementation('org.apache.commons:commons-compress:1.25.0') {
+            because '1.22 has security vulnerabilities'
         }
```

Comment:
This is a transitive dependency from https://mvnrepository.com/artifact/io.kubernetes/client-java -- we are currently on 18.0.0 but 19.0.0 is available and fixes this vulnerability.

I would remove this constraint (and with it, you could get rid of the whole constraints block) and instead update our Kubernetes client dependency:

```
    implementation 'io.kubernetes:client-java:19.0.0'
```

When I do that, here's what I get for a dependency insight:
```
(base) okotsopo@wm111-e35 jade-data-repo % ./gradlew dependencyInsight --dependency commons-compress

> Task :dependencyInsight
org.apache.commons:commons-compress:1.24.0
   variant "compile" [
      org.gradle.status              = release (not requested)
      org.gradle.usage               = java-api
      org.gradle.libraryelements     = jar (compatible with: classes)
      org.gradle.category            = library

      Requested attributes not found in the selected variant:
         org.gradle.dependency.bundling = external
         org.gradle.jvm.environment     = standard-jvm
         org.gradle.jvm.version         = 17
   ]

org.apache.commons:commons-compress:1.24.0
\--- io.kubernetes:client-java:19.0.0
     \--- compileClasspath

```

Reply:
Thanks for the suggestion! Updating the kubernetes java-client broke something so I'll look into that a bit more and see what changes we need to make to use that newer version!

Reply:
I looked into this issue and it turns out the terra common lib uses version `16.0.0` of this library and when I update TDR to use `19.0.0`, I get this error because that version adds an extra parameter to some methods: 
> An attempt was made to call a method that does not exist. The attempt was made from the following location: bio.terra.common.kubernetes.KubeService.getPodSet(KubeService.java:107). The following method did not exist: 'io.kubernetes.client.openapi.models.V1PodList
io.kubernetes.client.openapi.apis.CoreV1Api.listNamespacedPod(java.lang.String, java.lang.String, java.lang.Boolean, java.lang.String, java.lang.String, java.lang.String, java.lang.Integer, java.lang.String, java.lang.String, java.lang.Integer, java.lang.Boolean)'

I'll leave it at 18.0.0 for now if that's okay since this is currently blocked

Reply:
Great find, I appreciate you digging further into this.  Leaving at 18.0.0 sounds good.

Would you be willing to move this constraint under our kubernetes import with a comment indicating that it can be removed when we're on version >= 19.0.0, and a high-level note on why we can't upgrade it yet?